source/common/config/utility.h

```diff
@@ -531,7 +531,23 @@ class Utility {
     }
     return std::make_unique<FixedBackOffStrategy>(dns_refresh_rate_ms);
   }
-};
 
+  /**
+   * Prepares Jittered Exponential BackOff Strategy from Retry Policy config
+   * @param config retry policy config
+   * @param random random generator
+   * @return JitteredExponentialBackOffStrategyPtr
+   */
+  static JitteredExponentialBackOffStrategyPtr prepareJitteredExponentialBackOffStrategy(
```

Comment:
I think it would be better to have a 2 functions: one that receives the random-generator and an optional RetryPolicy (OptRef<const>), and another that is just like the first but with additional default values (the first function calling the second).
This way the function could check whether there is a retry policy, and if not, use the given default values.

Reply:
Sorry, I am trying to confirm my understanding here. Are you referring to something like this?

 ```
static JitteredExponentialBackOffStrategyPtr prepareJitteredExponentialBackOffStrategy(
      OptRef<const envoy::config::core::v3::RetryPolicy> retry_policy,
      Random::RandomGenerator& random);
```
&

```
static JitteredExponentialBackOffStrategyPtr prepareJitteredExponentialBackOffStrategy(
      OptRef<const envoy::config::core::v3::RetryPolicy> retry_policy,
      Random::RandomGenerator& random, uint32_t retry_initial_delay_ms = 1000,
      uint32_t retry_max_delay_ms = 10000);
```